### ISSUE TITLE
ios 15 SwiftUI stuttering problem

### ISSUE THREAD

**Issue by misimbits** [airbnb/lottie-ios#1414]
## This issue is a:
- Non-Crashing Bug (Visual or otherwise)

## Which Version of Lottie are you using?
Lottie 3.2.3

## What Platform are you on? 

- iOS

## What Language are you in?
- Swift (SwiftUI)

## Actual Behavior

After updating from ios 14 to 15, there was an animation stutter. It usually causes problems when pressing buttons that are not related to animation, and it works normally again after stuttering for about 1 to 2 seconds.

++
I found out that unlike Button, onTapGesture has no stuttering. And as a result of running the time profiler, I confirmed that displayLink.asyncThread occurs only in button, which affects the performance.

## Code Example
<img width="645" alt="스크린샷 2021-11-08 오후 12 20 06" src="https://user-images.githubusercontent.com/47140934/140679193-d5b71234-9d7e-458b-b776-fe2fb6f63894.png">
<img width="322" alt="스크린샷 2021-11-08 오후 6 10 13" src="https://user-images.githubusercontent.com/47140934/140714491-21552ab4-e61b-419c-b517-c2a87c20c5ae.png">

## Animation JSON

This happens on basically all Lottie animations

**Comment by brandenlivermore** [comment]
I am also experiencing this issue—animating the opacity of unrelated views causes the Lottie animation view to stutter.

**Comment by minlite** [comment]
+1.

I have a simple lottie loader animation running in an AnimationView. Smooth on iOS 14 (tested iPhone X and iPhone 12 Pro), laggy with frame drops on iOS 15 (iPhone 12 Pro). 

Lottie version: 3.2.3

**Comment by mkchoi212** [comment]
+1 on this issue

**Comment by misimbits** [comment]
> I experienced the same laggy issue when going from iOS 14 to iOS 15.
> 
> The problem is not only with buttons but with **all native animations in general**. In your case (@misimbits), when a button is pressed, there is a small native animation in place (so the user sees the feedback) and due to this animation, the laggy effect appears.
> 
> In my case, I wanted to have another simple animation in a combination with Lottie animation on the same view and the result was the same - laggy.
> 
> Lottie team, please fix this issue. Thanks! :)

**Comment by simba909** [comment]
We have encountered this in a project I work on as well, and it _seems_ to have been fixed as of iOS 15.2 (beta 3). Might be worth investigating for others seeing the same issue 👀

**Comment by schornon** [comment]
+1.
iOS 14 - ok.
iOS 15.0.2 - lagging, but can fix using `onTapGesture {}`:
```swift
Text("Start")
    .onTapGesture {
        isPlaing = true
}
```

**Comment by minlite** [comment]
@calda would you be able to look at this issue? We are seeing a significant performance regression in iOS 15 with lottie animations.

**Comment by calda** [comment]
Could somebody share a sample project that reproduces this issue? Is this only happening with SwiftUI?

**Comment by minlite** [comment]
Thanks for the reply! Trying to reproduce it in a minimal project. And no, in my case I was using UIKit.

**Comment by lovepreetSinghElectricpe** [comment]
It has been fixed in latest iOS 15.2 update.

**Comment by calda** [comment]
Today we released Lottie 3.4.0 which includes a new rendering engine powered by Core Animation ([more details here](https://github.com/airbnb/lottie-ios/discussions/1627)). This new rendering engine has significantly better performance, and should resolve the performance issues folks have been encountering with the previous main thread rendering engine. Please try it out by setting the `LottieConfiguration.renderingEngine` feature flag, and feel free to report any new issues that you experience when using the new rendering engine.
